schema-engine/sql-migration-tests/tests/query_introspection/mysql.rs

```diff
@@ -0,0 +1,517 @@
+        string(ScalarType::String) => (Bytes, Text),
+        bigint(ScalarType::BigInt) => (Bytes, Int64),
+        float(ScalarType::Float) => (Bytes, Double),
+        bytes(ScalarType::Bytes) => (Bytes, Bytes),
+        bool(ScalarType::Boolean) => (Bytes, Int32),
+        dt(ScalarType::DateTime) => (Bytes, DateTime),
+        decimal(ScalarType::Decimal) => (Bytes, Numeric),
+    );
+
+    test_native_types! {
+        Mysql57;
+
+        int(MySqlType::Int) => (Bytes, Int32),
+        unsigned_int(MySqlType::UnsignedInt) => (Bytes, Int64),
+        small_int(MySqlType::SmallInt) => (Bytes, Int32),
+        unsigned_small_int(MySqlType::UnsignedSmallInt) => (Bytes, Int32),
+        tiny_int(MySqlType::TinyInt) => (Bytes, Int32),
+        unsigned_tiny_int(MySqlType::UnsignedTinyInt) => (Bytes, Int32),
+        medium_int(MySqlType::MediumInt) => (Bytes, Int32),
+        unsigned_medium_int(MySqlType::UnsignedMediumInt) => (Bytes, Int64),
+        big_int(MySqlType::BigInt) => (Bytes, Int64),
+        decimal(MySqlType::Decimal(Some((4, 4)))) => (Bytes, Numeric),
+        unsigned_big_int(MySqlType::UnsignedBigInt) => (Bytes, Int64),
+        float(MySqlType::Float) => (Bytes, Float),
+        double(MySqlType::Double) => (Bytes, Double),
+        bit(MySqlType::Bit(1)) => (Bytes, Boolean),
+        char(MySqlType::Char(255)) => (Bytes, Text),
+        var_char(MySqlType::VarChar(255)) => (Bytes, Text),
+        binary(MySqlType::Binary(255)) => (Bytes, Bytes),
+        var_binary(MySqlType::VarBinary(255)) => (Bytes, Bytes),
+        tiny_blob(MySqlType::TinyBlob) => (Bytes, Bytes),
+        blob(MySqlType::Blob) => (Bytes, Bytes),
+        medium_blob(MySqlType::MediumBlob) => (Bytes, Bytes),
+        long_blob(MySqlType::LongBlob) => (Bytes, Bytes),
+        tiny_text(MySqlType::TinyText) => (Bytes, Text),
+        text(MySqlType::Text) => (Bytes, Text),
+        medium_text(MySqlType::MediumText) => (Bytes, Text),
+        long_text(MySqlType::LongText) => (Bytes, Text),
+        date(MySqlType::Date) => (Bytes, Date),
+        time(MySqlType::Time(Some(3))) => (Bytes, Time),
+        date_time(MySqlType::DateTime(Some(3))) => (Bytes, DateTime),
+        timestamp(MySqlType::Timestamp(Some(3))) => (Bytes, DateTime),
+        year(MySqlType::Year) => (Bytes, Int32),
+        json(MySqlType::Json) => (Bytes, Json),
+    }
+}
+
+mod mysql56 {
+    use super::*;
+
+    test_scalar_types!(
+        Mysql56;
+
+        int(ScalarType::Int) => (Bytes, Int32),
+        string(ScalarType::String) => (Bytes, Text),
+        bigint(ScalarType::BigInt) => (Bytes, Int64),
+        float(ScalarType::Float) => (Bytes, Double),
+        bytes(ScalarType::Bytes) => (Bytes, Bytes),
+        bool(ScalarType::Boolean) => (Bytes, Int32),
+        dt(ScalarType::DateTime) => (Bytes, DateTime),
+        decimal(ScalarType::Decimal) => (Bytes, Numeric),
+    );
+
+    test_native_types! {
+        Mysql56;
+
+        int(MySqlType::Int) => (Bytes, Int32),
```

Comment:
Should we maybe, for mysql 5.6 & 5.7, force param types to be unknown instead of `Bytes`

Reply:
I added this change. Lemme know what you think.